### PR TITLE
feature: add channel weights to discord plugin

### PR DESCRIPTION
**Pull request by crisog**
This PR builds on #2390. It adds the functionality to set weights to channels, along with a default property for the ones that are not weighted in the config.

I ran this plugin on the current 1Hive instance, sanity checked with @befitsandpiper and all tests are passing.

### CODE REVIEW

**Review by topocount — COMMENTED** [review]
@crisog please rebase on master and unmerge all the role-weight code when you have a chance. It'll make this easier to review. Happy to help if you like

**Comment by crisog** [comment]
> @crisog please rebase on master and unmerge all the role-weight code when you have a chance. It'll make this easier to review. Happy to help if you like

Sure! It should be fine now. Let me know if I need to do anything else @topocount.

File: src/plugins/experimental-discord/plugin.js
Comment (topocount) [inline src/plugins/experimental-discord/plugin.js:86]: Let's use NullUtil here: https://github.com/sourcecred/sourcecred/blob/d45ba0924c020ffecea0f3e0c866ac0b9c192211/src/util/null.js#L79-L81

File: src/plugins/experimental-discord/createGraph.js
Comment (topocount) [inline src/plugins/experimental-discord/createGraph.js:252]: NullUtil could be used here as well

File: src/plugins/experimental-discord/createGraph.js
Comment (topocount) [inline src/plugins/experimental-discord/createGraph.js:258]: :star_struck:

**Review by topocount — APPROVED** [review]
@crisog nice work! Just a couple nits to start bringing this plugin more in line with the rest of the codebase.

I also want to wait for approval from @decentralion and/or @wchargin before merging

File: src/plugins/experimental-discord/config.js
Comment (befitsandpiper) [inline src/plugins/experimental-discord/config.js:41]: nit: replace "role" with "channel"

**Review by befitsandpiper — APPROVED** [review]

File: src/plugins/experimental-discord/plugin.js
Comment (decentralion) [inline src/plugins/experimental-discord/plugin.js:87]: I don't think this will quite work, but I'll resolve it in another PR, since it's a pre-existing issue.

**Review by decentralion — APPROVED** [review]
Looks good, except for the issue of loading a config that doesn't have this explicitly set, but that issue looks to have been introduced in an earlier PR so it doesn't need to change in this one.

File: src/plugins/experimental-discord/plugin.js
Comment (decentralion) [reply src/plugins/experimental-discord/plugin.js:87]: Nevermind! It's fine. :)

**Pull request merged**
